### DISCUSSION TITLE
Anyway to skip http validation?

### DISCUSSION THREAD

**Discussion by stellyuk** [nginx-proxy/acme-companion#833] [Q&A] [answered]
Hey,

I use http for a different service and was wondering if there was any way that I could get it to validate a cert over port 443 (ssl) or any other port/way?

Thanks!

**Comment by buchdag** [comment, marked as answer]
Hi. Unfortunately no, `http-01` challenge requires port `80` (this on the ACME standard, there is nothing we can do about it) and for now this is the only challenge that can be used with this container.

https://letsencrypt.org/docs/challenge-types/#http-01-challenge

> The HTTP-01 challenge can only be done on port 80. Allowing clients to specify arbitrary ports would make the challenge less secure, and so it is not allowed by the ACME standard.

https://letsencrypt.org/docs/allow-port-80/

**Reply by stellyuk** [reply]
Thanks, I thought as much, just need to look at a different solution but love the whole project :)

**Comment by polarathene** [comment]
`acme-companion` uses `acme.sh` to handle the challenges. Currently it only supports HTTP-01 challenges over port 80, however `acme.sh` has support for TLS-ALPN-01 challenge (HTTPS port 443) and DNS-01 challenge (_no ports necessary, but depends on your DNS provider support_, also required for wildcard certificates).

In future, `acme-companion` could support the other challenges, but someone would need to contribute the integration with `acme.sh` features.

---

You could instead provision your certificates via `certbot` or `acme.sh` directly AFAIK, and provide the certs to `nginx-proxy`.

Generally though, you should be using `nginx-proxy` to port HTTP port 80 to any other service that would use HTTP. This can be port 80 within a different container or network, or if the service allows using a different port for HTTP, proxying the web port 80 that `nginx-proxy` manages to that services port and everything should work smoothly for you.
